Failed on uploading a large (1GB+) file in Releases

**Describe the bug**
When uploading a large file in Releases (after raising the default limit), the upload times out with the error message:

`[ERROR] [reflect/value.go:460 call()] parse multipart form: multipart: NextPart: stream error: stream ID 39; CANCEL`

or, in a similar vein:

`[reflect/value.go:460 call()] parse multipart form: multipart: NextPart: stream error: stream ID 81; CANCEL`

**Gogs version and commit**
Build time: 2020-05-04 10:53:31 UTC
Build commit: 9fd4f5562d7a59abbef5e78d44fb85a0041e0733

**Git version**

```
2.24.3
```

**Operating system**
Ubuntu 16.04 LTS (docker gogs/gogs:latest)

**Database**
postgres 9.5 (docker)

**To Reproduce**
Steps to reproduce the behavior:
1. Raise release file size limit. 
2. Upload at least a 1.5GB+ file as a Release in a repo.
3. Wait for it to timeout.

**Can you reproduce the bug at https://try.gogs.io?**
No, cannot upload file that large.

**Expected behavior**
I expect Gogs to complete the file upload and not time out.

**Actual behavior**
Gogs will time out, produce the error code above, and not allow the download to continue.

**Screenshots**
![image](https://user-images.githubusercontent.com/2816245/81183566-4c2e2d00-8f7d-11ea-88b1-3db13981c9e9.png)


**Additional context**
```
2020/05/06 13:23:42 [ INFO] Run mode: Production
2020/05/06 13:23:43 [ INFO] Listen on https://0.0.0.0:3000
2020/05/06 13:24:29 [ERROR] [reflect/value.go:460 call()] parse multipart form: multipart: NextPart: stream error: stream ID 39; CANCEL
2020/05/06 13:28:01 [ERROR] [reflect/value.go:460 call()] parse multipart form: multipart: NextPart: stream error: stream ID 81; CANCEL
```

I've tried both Chrome and Firefox. Uploading from Windows Server 2016.

--- a/public/js/gogs.js
+++ b/public/js/gogs.js
@@ -1474,6 +1474,7 @@ $(document).ready(function() {
       headers: { "X-CSRF-Token": csrf },
       maxFiles: $dropzone.data("max-file"),
       maxFilesize: $dropzone.data("max-size"),
+      timeout: 0,
       acceptedFiles:
         $dropzone.data("accepts") === "*/*" ? null : $dropzone.data("accepts"),
       addRemoveLinks: true,

